internal/docker/deployment.go

```diff
@@ -80,6 +82,24 @@ func (d *Deployment) Client(t *testing.T, hsName, userID string) *client.CSAPI {
 	return client
 }
 
+// Random hex-encoded string function, needed for NewUser, taken from https://stackoverflow.com/a/65607935/8700553
+func randomString(length int) string {
+	rand.Seed(time.Now().UnixNano())
+	b := make([]byte, length)
+	rand.Read(b)
+	return fmt.Sprintf("%x", b)[:length]
+}
+
+// NewUser creates a new user as a convenience method to RegisterUser.
+//
+//It registers the user with a random password, and without admin privileges.
+func (d *Deployment) NewUser(t *testing.T, localpart, hs string) *client.CSAPI {
+	// Dendrite requires at least 8 characters for a password, we choose 12 here in case they increase that.
+	password := randomString(12)
```

Comment:
Please don't do this. We don't need the security for these accounts. Make them deterministic to make it easier for developers to introspect servers. For example, I've seen cases where developers keep the HS containers running then point a local element web at it in order to check what is on the account - they need to login with a password and can do so because it is deterministic. Making this random makes this impossible.

Reply:
> Use the same deterministic code as https://github.com/matrix-org/complement/blob/main/internal/instruction/runner.go#L545 please.

Thanks for mentioning that, I'll obviously do that, though I wasn't aware of that bit of code before 👍